google-map-marker.html

```diff
@@ -270,20 +288,35 @@
         if (!this.info) {
           // Create a new infowindow
           this.info = new google.maps.InfoWindow();
-          this.infoHandler_ = google.maps.event.addListener(this.marker, 'click', function() {
-            this.info.open(this.map, this.marker);
+          this.openInfoHandler_ = google.maps.event.addListener(this.marker, 'click', function() {
```

Comment:
All DOM events can be fired from `_openChanged` and this `click` and the `closeclick` handlers can just set `this.open` correctly.